src/filesystem/implementation.js

```diff
@@ -1249,6 +1249,9 @@ function truncate_file(context, path, length, callback) {
       if(!fileData) {
         return callback(new Errors.EIO('Expected Buffer'));
       }
+      if (!Number.isInteger(length)) {
```

Comment:
node does this: https://github.com/nodejs/node/blob/master/lib/internal/validators.js#L62-L76  Do you want to just add that function in here and call it?  There might be a few other places that need it.

Reply:
Great point - do you think we should put these validation functions into a separate file so that other parts of the codebase can use it?

Reply:
Throw it in `shared.js`